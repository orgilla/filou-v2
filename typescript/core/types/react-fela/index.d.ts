declare module "react-fela" {
  export const FelaComponent: any;
  export const FelaTheme: any;
}
