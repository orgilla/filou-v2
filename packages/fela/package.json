{
  "version": "0.14.8",
  "name": "@orgilla/fela",
  "description": "",
  "main": "lib",
  "modules": "es",
  "author": "Orgilla <hi@orgil.la>",
  "contributors": [
    "Benjamin Kniffler <bkniffler@me.com>",
    "Philipp Achenbach <olymp@philipp-achenbach.de>"
  ],
  "license": "MIT",
  "dependencies": {
    "fela": "^6.2.1",
    "fela-plugin-custom-property": "^7.0.3",
    "fela-plugin-embedded": "^5.2.11",
    "fela-plugin-extend": "^6.0.9",
    "fela-plugin-fallback-value": "^5.0.18",
    "fela-plugin-friendly-pseudo-class": "^5.0.17",
    "fela-plugin-named-keys": "^1.0.3",
    "fela-plugin-prefixer": "^5.0.19",
    "fela-plugin-unit": "^5.0.17",
    "tinycolor2": "^1.4.1"
  },
  "peerDependencies": {
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "react-fela": "^8.0.0"
  }
}
