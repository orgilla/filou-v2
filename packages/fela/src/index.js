export { default as withTheme } from './with-theme';
export { default as withPulse } from './with-pulse';
export { default as colorsMaterial } from './colors-material';
export { default as ColorsProvider, getColor } from './colors-provider';

export { createComponent, Provider } from 'react-fela';
export { default as color } from 'tinycolor2';
export { default as tinyColor } from 'tinycolor2';

export { default as createRenderer } from './create-renderer';
export { default as ThemeProvider } from './theme-provider';
export { default as normalize } from './normalize';
export { default as withStyle } from './with-style';
