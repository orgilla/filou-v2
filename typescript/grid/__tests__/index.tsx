import 'jest';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createRenderer } from 'fela';
import { Provider as FelaProvider, ThemeProvider } from 'react-fela';
import { renderToMarkup } from 'fela-dom';
import Grid from '../src';

const getProvider = (fela: any, theme = {}) => {
  const Provider: React.SFC = ({ children }) => (
    <FelaProvider rehydrate renderer={fela}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FelaProvider>
  );
  return Provider;
};

describe('Grid', () => {
  it('should render correctly', () => {
    const fela = createRenderer();
    const Provider = getProvider(fela);
    const component = renderer.create(
      <Provider>
        <Grid size={12}>
          <Grid.Item mini={6}>hi</Grid.Item>
          <Grid.Item mini={6}>hi</Grid.Item>
        </Grid>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot();
  });
});
