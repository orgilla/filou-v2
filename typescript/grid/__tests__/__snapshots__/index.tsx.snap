// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `
<div
  className="a"
>
  <div
    className="b c d"
  >
    hi
  </div>
  <div
    className="b c d"
  >
    hi
  </div>
</div>
`;

exports[`Grid should render correctly 2`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"4\\" data-fela-type=\\"RULE\\">.a{min-width:100%}.b{float:left}.c{width:50%}.d{margin-left:0%}</style>"`;
